test/cli-tests/test-block.js

```diff
@@ -1,61 +1,75 @@
 /* eslint-env mocha */
+'use strict'
 
 const expect = require('chai').expect
 const nexpect = require('nexpect')
 const httpAPI = require('../../src/http-api')
+const path = require('path')
+const repoPath = require('./index').repoPath
+
+const spawn = (args) => {
+  const env = process.env
+  env.IPFS_PATH = repoPath
+  return nexpect.spawn(
+    'node',
+    [path.join(__dirname, '../../src/cli/bin.js')].concat(args),
+    {env}
```

Comment:
This is not setting properly the `IPFS_PATH` for the spawned process (if you print it from https://github.com/ipfs/js-ipfs/blob/master/src/cli/utils.js#L46-L48) you will see it is undefined.

This is causing the cli process to fallback to default dir, finding that the API might be accessible through 8080, dialling to another API instead of the one generated for their tests.